app/frontend/src/utils/date.ts

```diff
@@ -1,6 +1,11 @@
 // format a date
+import { Timestamp } from "google-protobuf/google/protobuf/timestamp_pb";
 import moment from "moment";
 
 export function formatDate(date: string): string {
   return moment(date, "YYYY-MM-DD").format("D MMM YYYY");
 }
+
+export function timestamp2Date(timestamp: Timestamp.AsObject): Date {
+  return new Date(Math.floor(timestamp.seconds * 1e3 + timestamp.nanos / 1e6));
```

Comment:
Why is there a `Math.floor`? Here's the `toDate` function from the proto:

https://github.com/protocolbuffers/protobuf/blob/d16bf914bc5ba569d2b70376051d15f68ce4322d/src/google/protobuf/compiler/js/well_known_types_embed.cc#L99-L104

Reply:
Because the spec says it should be an integer: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date

Reply:
Ahh, okay; that makes sense. So it's truncating the nanos a bit, to milliseconds.